tests/validator.go

```diff
@@ -0,0 +1,277 @@
+// Copyright 2017 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package blackbox
+
+import (
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"syscall"
+	"testing"
+
+	"github.com/coreos/ignition/internal/exec/util"
+	"github.com/coreos/ignition/tests/types"
+)
+
+func validatePartitions(t *testing.T, expected []*types.Partition, imageFile string) {
+	for _, e := range expected {
+		if e.TypeCode == "blank" {
+			continue
+		}
+		sgdiskInfo, err := exec.Command(
+			"sgdisk", "-i", strconv.Itoa(e.Number),
+			imageFile).CombinedOutput()
+		if err != nil {
+			t.Error("sgdisk -i", strconv.Itoa(e.Number), err)
+			return
+		}
+
+		re, err := regexp.Compile("Partition unique GUID: (?P<partition_guid>[\\d\\w-]+)")
```

Comment:
`regexp.MustCompile` would be appropriate here.

The repeated code could move into a helper function.